k-distribution/tests/regression-new/checks/checkBracket.k

```diff
@@ -0,0 +1,10 @@
+module CHECKBRACKET-SYNTAX
+  syntax {A} A ::= "(" A ")" [bracket]
+  syntax Exp ::= "(" Exp ")" [bracket]
+  syntax Exp ::= "(" Exp Exp ")" [bracket]
+endmodule
```

Comment:
Please also add some examples of bracket productions that are ill-formed in the following ways:

-   The thing inside the bracket is a subsort of the final sort.
-   The thing inside the bracket is a supersort of the final sort.
-   The thing inside the bracket is an unrelated sort to the final sort.

Also add these tests which should not be errors:

-  The bracket is not a surrounding production like above, but some weird left only, right only, or mixed operator. Add a few of these.
-  The bracket is not traditional bracket characters like `(` or `[`, but something ilke `%`.